### PR TITLE
feat(medusa): Trim discount code on insert and retrieve

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/medusa/src/services/discount.ts [hunk 1]
```diff
@@ -277,7 +275,7 @@ class DiscountService extends TransactionBaseService {
     const manager = this.manager_
     const discountRepo = manager.getCustomRepository(this.discountRepository_)
 
-    const normalizedCode = discountCode.toUpperCase()
+    const normalizedCode = discountCode.toUpperCase().trim()
```
Comment (olivermrbl) [inline packages/medusa/src/services/discount.ts:278]: comment: This is on `retrieveByCode`

File: packages/medusa/src/services/discount.ts [hunk 2]
```diff
@@ -212,8 +212,6 @@ class DiscountService extends TransactionBaseService {
         const discountRule = ruleRepo.create(validatedRule)
         const createdDiscountRule = await ruleRepo.save(discountRule)
 
-        discount.code = discount.code!.toUpperCase()
-
         const created: Discount = discountRepo.create(
```
Comment (olivermrbl) [inline packages/medusa/src/services/discount.ts:215]: comment: Removed from `create` as we are using `beforeInsert` in the model

Comment (srindom) [inline packages/medusa/src/services/discount.ts:215]: question: why is this removed?

Comment (srindom) [reply packages/medusa/src/services/discount.ts:215]: Sorry didnt see your comment

**Review by srindom — APPROVED** [review]
LGTM!

**Review by pKorsholm — APPROVED** [review]
lgtm! 💪
